src/internal/connector/onedrive/restore.go

```diff
@@ -157,29 +159,29 @@ func RestoreCollection(
 		return metrics, clues.Wrap(err, "creating drive path").WithClues(ctx)
 	}
 
-	if rootIDCache == nil {
-		rootIDCache = map[string]string{}
+	if rootFolderIDCache == nil {
+		rootFolderIDCache = map[string]string{}
 	}
```

Comment:
this seems like a code smell. Either `rootFolderIDCache` doesn't need to be an in-out param, and thus doesn't need to be passed in as a parameter at all (it's not returned), or it _is_ an in-out param and stuff is could break in weird ways if nil is passed in

Reply:
This gets resolved in the next PR.